internal/server/edit_complaint.go

```diff
@@ -64,23 +67,36 @@ func EditComplaint(client EditComplaintClient, tmpl template.Template) Handler {
 		}
 
 		if r.Method == http.MethodPost {
-			complaint := sirius.Complaint{
-				Category:             postFormString(r, "category"),
-				Description:          postFormString(r, "description"),
-				ReceivedDate:         postFormDateString(r, "receivedDate"),
-				Severity:             postFormString(r, "severity"),
-				InvestigatingOfficer: postFormString(r, "investigatingOfficer"),
-				ComplainantName:      postFormString(r, "complainantName"),
-				SubCategory:          getValidSubcategory(postFormString(r, "category"), r.PostForm["subCategory"]),
-				ComplainantCategory:  postFormString(r, "complainantCategory"),
-				Origin:               postFormString(r, "origin"),
-				CompensationType:     postFormString(r, "compensationType"),
-				Summary:              postFormString(r, "summary"),
-				Resolution:           postFormString(r, "resolution"),
-				ResolutionInfo:       postFormString(r, "resolutionInfo"),
-				ResolutionDate:       postFormDateString(r, "resolutionDate"),
+			resolution := postFormString(r, "resolution")
+
+			if resolution != "" {
+				fieldErrors := sirius.FieldErrors{}
+
+				for _, field := range fieldsToBeValidated {
+					if !isFieldPopulated(field, r) {
+						fieldErrors[field] = map[string]string{
+							"reason": "Value is required and can't be empty",
+						}
+					}
+				}
+
+				if len(fieldErrors) > 0 {
+					w.WriteHeader(http.StatusBadRequest)
+					data.Error.Field = fieldErrors
+					data.Complaint = populateComplaint(r)
+					return tmpl(w, data)
+				}
+			}
+
+			category := postFormString(r, "category")
+			if category != "" {
+				if getValidSubcategory(postFormString(r, "category"), r.PostForm["subCategory"]) == "" {
+					return getSubcategoryValidationError(w, r, tmpl, data)
+				}
```

Comment:
This is all fine but ideally we could just send the data and let the API handle the validation. Im wondering if we could create a custom validator in sirius API that would be able to handle all of this logic to save us having to do it in the frontend. I'm going to have a look into seeing if its a possibility to check the resolution state and conditionally apply validation off that

Reply:
I think this is possible in the `UpdateComplaintHandler` i'll give it a go

Reply:
Agree that this should be in the API so it can't be bypassed. If conditional validation is a PITA you could have two validators and switch them at the controller:
```php
$validator = $data['resolutionDate'] ? ResolveComplaint::class : UpdateComplaint::class;

$complaint = $this->commandBus($this->validatePayload($validator, $data));
```
I think we've done this somewhere already but I can't remember where.